lib/APMSpan.js

```diff
@@ -65,7 +73,43 @@ class APMSpan {
             nodeType = 'Consumer';
         }
 
-        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId);
+        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId ? [{
+            value: consumerCorrelationId,
+            scope: 'Interaction',
+        }] : undefined);
+    }
+
+    _initFollowsFromSpanContext(parents) {
+        function findRoot(span) {
+            let rootSpan = span;
+            while (rootSpan.getParentId()) {
+                const node = span.getTrace().findNode(rootSpan.getParentId());
+                if (node) {
+                    rootSpan = node.getSpan();
+                } else {
+                    break;
+                }
+            }
+
+            return rootSpan;
+        }
+
+        const id = utils.generateSpanId();
+        let traceId = null;
+
+        const correlationIds = [];
+        for (let i = 0; i < parents.length; i += 1) {
+            const parent = parents[i] instanceof APMSpan ? parents[i].context() : parents[i];
+            traceId = parent.getTraceId();
+            const nodePosition = parent.getTrace().nodePositionId(parent.getId());
+            correlationIds.push({
+                value: `${findRoot(parent).getId()}:${nodePosition}`,
+                scope: 'CausedBy',
+            });
+        }
+
+        this[SPAN_CONTEXT] = new APMSpanContext(id, traceId, null);
+        this[SPAN_CONTEXT].getTrace().addNode('Component', this, correlationIds);
```

Comment:
If FollowsFrom ref, then the type should be `Consumer` with endpointType = null to represent an internal link.